### PR TITLE
monero-cli: fix build issue with GCC14

### LINKED ISSUE #367857: Build failure: monero-cli
## Steps To Reproduce

1. build `monero-cli` on current master (7dcb1a29b6da3524f4ae5499f5c348dde40a711a)

## Build log

<details>

<summary>Build Log</summary>

```
[...]
monero-cli> /build/source/src/rpc/core_rpc_server.cpp:3270:0:   required from here
monero-cli>  3270 |       const std::string span_connection_id = epee::string_tools::pod_to_hex(span.connection_id);
monero-cli> /build/source/contrib/epee/include/span.h:165:34: error: static assertion failed: source type may have padding
monero-cli>   165 |     static_assert(!has_padding<T>(), "source type may have padding");
monero-cli>       |                    ~~~~~~~~~~~~~~^~
monero-cli> /build/source/contrib/epee/include/span.h:165:34: note: '! epee::has_padding<boost::uuids::uuid>()' evaluates to false
monero-cli> make[2]: *** [src/rpc/CMakeFiles/obj_rpc.dir/build.make:104: src/rpc/CMakeFiles/obj_rpc.dir/core_rpc_server.cpp.o] Error 1
monero-cli> make[1]: *** [CMakeFiles/Makefile2:2114: src/rpc/CMakeFiles/obj_rpc.dir/all] Error 2
monero-cli> make: *** [Makefile:146: all] Error 2
```

</details>

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.64, NixOS, 25.05 (Warbler), 25.05.20241212.2f91aa2`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(wogan): `""`
 - channels(root): `"nixos, nixpkgs"`
 - nixpkgs: `/nix/store/gffj8gl72kb46j9k5g1yg1qc5jrzb7b4-source`

## Notify maintainers

@philipmw @rnhmjoj

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by gador**
fix build with GCC14 update

fixes #367857

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by gador** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367932`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>haven-cli</li>
    <li>haven-cli.source</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>atomic-swap</li>
    <li>monero-cli</li>
    <li>monero-cli.source</li>
    <li>monero-gui</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>haven-cli</li>
    <li>haven-cli.source</li>
  </ul>
</details>

**Comment by gador** [comment]
`haven-cli` fails on master, too

File: pkgs/applications/blockchains/monero-cli/default.nix [hunk 1]
```diff
@@ -53,7 +52,11 @@ stdenv.mkDerivation rec {
     hash = "sha256-nDiFJjhsISYM8kTgJUaPYL44iyccnz5+Pd5beBh+lsM=";
   };
 
-  patches = [ ./use-system-libraries.patch ];
+  patches = [
+    ./use-system-libraries.patch
+    # https://github.com/monero-project/monero/pull/9462
```
Comment (rnhmjoj) [inline pkgs/applications/blockchains/monero-cli/default.nix:57]: Can you please use `fetchpatch` to download the [merged commit](https://github.com/monero-project/monero/commit/83dd5152e6d115426afbb57a94a832ec91b58a46.patch)?

Comment (gador) [reply pkgs/applications/blockchains/monero-cli/default.nix:57]: Sure!

**Review by rnhmjoj — APPROVED** [review]
I did the usual test: connect monero-gui to a remote node. Looks good.

**Pull request merged**